src/fu-engine.c

```diff
@@ -5771,6 +5771,52 @@ fu_engine_ensure_security_attrs_supported (FuEngine *self)
 	}
 }
 
+static gchar *
+fu_engine_attrs_calculate_hsi_for_chassis (FuEngine *self)
+{
+	guint val;
+	g_autoptr(GError) error = NULL;
+
+	/* get chassis type from SMBIOS data */
+	val = fu_smbios_get_integer (self->smbios,
+				     FU_SMBIOS_STRUCTURE_TYPE_CHASSIS,
+				     0x05, &error);
+	if (val == G_MAXUINT) {
+		g_warning ("failed to get chassis type: %s", error->message);
+		return g_strdup ("HSI-INVALID:chassis");
+	}
```

Comment:
can't this be part of the switch/case?

Reply:
I think i'd rather handle failure with something setting the GError before we get into the "filtering" switch section.